lib/job.js

```diff
@@ -133,6 +133,14 @@ Job.prototype.toData = function() {
   const json = this.toJSON();
 
   json.data = JSON.stringify(json.data);
+
+  if (
+    this.opts.sizeLimit &&
+    utils.lengthInUtf8Bytes(json.data) > this.opts.sizeLimit
+  ) {
+    throw new Error(`The size of job ${this.opts.jobId || this.name || Job.DEFAULT_JOB_NAME} exceeds the limit ${this.opts.sizeLimit} bytes`);
```

Comment:
Since toData can throw now, are there any other callers of `toData` that need to handle this?

Reply:
Also, `toData` seems like a weird place to check this. Wouldn't `addJob` be a better place?

Reply:
but even if it is called from other places, isn't it good that it throws if it exceeds the maxium size?

Reply:
I don't think so. If I want to serialize a job to JSON, I don't want that to throw if it's too big. I only care about it being too big when I add the job to the queue.

Reply:
I can make this change, thanks for the feedback.